### DISCUSSION TITLE
`getInstallPath()` of (possibly root) package

### DISCUSSION THREAD

**Discussion by mvorisek** [composer/composer#11927] [Q&A / Support] [answered]
Currently, `LibraryInstaller::getInstallPath()` does return wrong path when root package is given.

This is "expected" as the underlying code simply appends the package name to the vendor directory: https://github.com/composer/composer/blob/2.7.2/src/Composer/Installer/LibraryInstaller.php#L240

Is this desired? If yes, please advise how to get root package path from a method in a class extending `LibraryInstaller`.

My current (not working code) is:

```
class ExtensionInstaller extends LibraryInstaller
{
    protected function getRoundcubemailInstallPath(): string
    {
        $roundcubemailPackage = $this->composer
            ->getRepositoryManager()
            ->findPackage('roundcube/roundcubemail', '*');

        if ($roundcubemailPackage === null) {
            $rootPackage = $this->composer->getPackage();
            if ($rootPackage->getName() === 'roundcube/roundcubemail') {
                $roundcubemailPackage = $rootPackage;
            }
        }

        return $this->getInstallPath($roundcubemailPackage);
    }
}
```

Also, what is the definition of `RepositoryManager::findPackage()` - will the code above work for `roundcube/roundcubemail` provided by some other package?

If no, how to figure out what package does provide the `roundcube/roundcubemail`?

If yes, is it safe to omit the `$rootPackage->getName() === 'roundcube/roundcubemail'` condition then or how to check if the root package replaces the `roundcube/roundcubemail`?

**Comment by Seldaek** [comment, marked as answer]
The root package is considered to be in `getcwd()`.

findPackage only matches on the name. If you want to include replace/provide you need to use an InstalledRepository and use findPackagesWithReplacersAndProviders
